### PR TITLE
Add authorization rules

### PR DESCRIPTION
**Pull request by MarcoDiaz2000**
## Description
In this project, we added authorization to our app using [CanCanCan](https://github.com/CanCanCommunity/cancancan).

### Project requirements
- Install CanCanCan in your project.
- Add a `role` column to the users table. Remember to use a migration for this.
- A user can delete a post if it is theirs or if they have an admin role (column `role` has value `"admin"`). Use CanCanCan for this authorization.
    - For that you need to implement the post deleting functionality. Add the "Delete" button to the view and make sure that only authorized users can see it.
- A user can delete a comment if it is theirs or if they have an admin role (column `role` has value `"admin"`). Use CanCanCan for this authorization.
    - For that you need to implement the comment deleting functionality. Add the "Delete" button to the view and make sure that only authorized users can see it.

### CODE REVIEW

File: app/controllers/posts_controller.rb [hunk 1]
```diff
@@ -1,5 +1,14 @@
 class PostsController < ApplicationController
   before_action :authenticate_user!, only: %i[new create]
+
+  def destroy
+    @post = Post.find(params[:id])
+    authorize! :destroy, @post
+    author = @post.author
+    @post.destroy
+    redirect_to user_posts_path(author), notice: 'Post was successfully deleted.'
```
Comment (ichala) [inline app/controllers/posts_controller.rb]: - [x] as part of the requirements , you need to use **CanCanCan** for Post authorization in `Post Controller` by adding   load_and_authorize_resource , kindly fix it 
```ruby
 class PostsController < ApplicationController
    load_and_authorize_resource

    ...
  end
```

**Review by ichala — COMMENTED** [review]
Hi Student
Good job so far! 
👏👏👏 
You still need to work on some issues to go to the next project, but you're almost there!

the-simpsons-homer-simpson

## To Highlight 🎉

- No Linter errors  ✔️ 
- Installed CanCanCan ✔️
- Has CanCanCan `ability` class that defines the set of rules for user✔️
- `role` column exist in the users table✔️


 ## Required Changes ♻️


- [ ] **_🛑  Check The Comments Below_**





## Optional suggestions

- N/A

**********
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you consider them as they can improve your code._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: app/controllers/posts_controller.rb (hunk 1 above)
Comment (MarcoDiaz2000) [reply app/controllers/posts_controller.rb]: Thanks

**Review by binhussen — APPROVED** [review]
Hi @MarcoDiaz2000 , 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉
IApproveSimonCowellGIF (4)

## Highlights
- Added authorization rule ✔️
- Well Structured Code✔️
- Used Git flow ✔️
## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**